.github/workflows/docker.yaml

```diff
@@ -0,0 +1,29 @@
+---
+
+name: Build Production Docker Image
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+    build-docker:
+        name: "Build Production Docker Image"
+        runs-on: "ubuntu-latest"
+        env:
+            DOCKER_LATEST: ${{ secrets.DOCKER_URL }}:latest
+            DOCKER_TAG_PRODUCTION: ${{ secrets.DOCKER_URL }}:${{ github.sha }}
+        steps:
+            - name: "Checkout repository"
+              uses: "actions/checkout@v2"
+
+            - name: "Build Docker image"
+              run: docker build --tag "${DOCKER_LATEST}" .
+
+            - name: "Login to GitHub"
+              run: echo "${{ secrets.GH_TOKEN }}" | docker login docker.pkg.github.com -u "${{ secrets.GH_USER }}" --password-stdin
```

Comment:
[https://github.community/t/how-do-i-get-gh-username-based-on-actions-events/17882/3](https://github.community/t/how-do-i-get-gh-username-based-on-actions-events/17882/3) seems like you can use `github.actor` which is way better